cmdstanpy/stanfit.py

```diff
@@ -1457,3 +1337,121 @@ def save_csvfiles(self, dir: str = None) -> None:
         :param dir: directory path
         """
         self.runset.save_csvfiles(dir)
+
+
+def from_csv(
+    dir: str = None, method: str = None
+) -> Union[CmdStanMCMC, CmdStanMLE, CmdStanVB]:
+    """
+    Given a directory of saved Stan CSV files and a method specification,
+    instantiate the appropriate object containing the CmdStan inference.
+
+    :param dir: directory path
+    :param method: CmdStan method name
+
+    :return: either a CmdStanMCMC, CmdStanMLE, or CmdStanVB object
+    """
+    if dir is None:
+        raise ValueError('Must specify directory of Stan CSV files.')
+    if not os.path.exists(dir):
+        raise ValueError('Directory {} not found.'.format(dir))
+    if method is None or method not in ['sample', 'optimize', 'variational']:
```

Comment:
I still believe it would be a good idea to allow method=None and figure it out from the files in this case. It would require moving some of the checks around, but I think it would be a good idea.

If we don't want to do that, I think we should remove the default assignments to None in the method signature. That will promote not supplying the arguments up to a syntax level rather than a valueerror level.

Reply:
makes sense.